### ISSUE TITLE
TextBox ignores HorizontalContent/TextAlignment APIs

### ISSUE THREAD

**Issue by Felix-Dev** [microsoft/microsoft-ui-xaml#2896]
**Describe the bug**
The TextBox control currently ignores the following two APIs to align its text inside of it:
* HorizontalContentAlignment
* HorizontalTextAlignment

See the following XAML markup:
```xml
<StackPanel>
    <Border BorderThickness="1" BorderBrush="Black" Margin="0,0,0,20">
        <TextBlock Text="TextBlock text" HorizontalTextAlignment="Right" Width="150" />
    </Border>
    
    <TextBox Text="TextBox text" Width="150"  HorizontalTextAlignment="Right" HorizontalContentAlignment="Right"/>
</StackPanel>
```

And the result:
![image](https://user-images.githubusercontent.com/1398851/87349935-6a08a700-c557-11ea-86d5-f275002c5e9f.png)

**Expected behavior**
Text of the TextBox should also be aligned to the right.

**Additional context**
This bug also affects all the controls using a TextBox internally, such as AutoSuggestBox and NumberBox. This is a gap in WinUI in regards to Win32 APIs like WinForms NumericUpDown control which can align its content to the right just fine:
![image](https://user-images.githubusercontent.com/1398851/87350363-0a5ecb80-c558-11ea-8ac7-95e5fdb698c3.png)

For the NumberBox, an own issue exists in https://github.com/microsoft/microsoft-ui-xaml/issues/1722 which would easily be done once this bug has been fixed.

**Version Info**

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| May 2020 Update (19041)            | Yes  |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Comment by MikeHillberg** [comment]
Can you try working around this by setting TextAlignment rather than HorizontalTextAlignment on the TextBox?

**Comment by Felix-Dev** [comment]
@MikeHillberg Not sure how I missed that property 😑

![image](https://user-images.githubusercontent.com/1398851/87366296-a3054380-c578-11ea-955c-1304c2452057.png)

**Comment by MikeHillberg** [comment]
No, I think it's still a bug; you should be able to use either property, your sample code looks valid.

I'm confused by the documentation too. It's saying that this is last-writer-wins behavior, but we have a pretty strict rule to not rely on the order that properties are set (too much ambiguity and too little control).

I don't remember how TextBox ended up with this duplicate property, but I think it was anticipating a VerticalTextAlignment property like Xamarin Forms has.

**Comment by Felix-Dev** [comment]
@MikeHillberg Yep, the documentation confused me for a second here which made me think this is a non-issue. Turns out there really still is an issue here. I wrote some test XAML to check when the `TextAlignment` and `HorizontalTextAlignment` APIs are applied. See the following XAML and code-behind.

```xml
<StackPanel Orientation="Horizontal" Margin="0,20,0,20" HorizontalAlignment="Center">
    <Button Content="Set HorizontalTextAlignment" Click="Button_Click" Margin="0,0,10,0"/>
    <Button Content="Set TextAlignment" Click="Button_Click_1"/>
</StackPanel>

<Border BorderThickness="1" BorderBrush="Black" Margin="0,20,0,20" Grid.Row="3" HorizontalAlignment="Center">
    <TextBlock Text="TextBlock text" HorizontalTextAlignment="Right" Width="150" />
</Border>

<TextBox x:Name="DemoTextBox" 
         Text="TextBox text" Width="150" Grid.Row="4" Margin="0,0,0,20" TextAlignment="Left" HorizontalTextAlignment="Left"/>
```
```cs
private void Button_Click(object sender, RoutedEventArgs e)
{
    this.DemoTextBox.HorizontalTextAlignment = TextAlignment.Right;
}

private void Button_Click_1(object sender, RoutedEventArgs e)
{
    this.DemoTextBox.TextAlignment = TextAlignment.Right;
}
```
This is how it plays out:
![textbox-textalignment](https://user-images.githubusercontent.com/1398851/87433733-f6fc4080-c5e9-11ea-9118-b149a4fc23e4.gif)

As you can see, setting `HorizontalTextAlignment` on the TextBox has *no effect* at all. That API simply appears to be ignored. 

This doesn't match the documentation for the `TextBox.HorizontalTextAlignment` API at all:
> Apps that target the Fall Creators Update (SDK 16299) or later should use this property instead of TextAlignment. 

There is clearly a disconnect between the documentation and the API behavior (I ran the example code above targeting the 18362 SDK). As it stands now, the `HorizontalTextAlignment` property is meaningless to set on at least Windows 10 1903 and above and we should figure out if this is a documentation issue or an API issue.

**Comment by StephenLPeters** [comment]
@chrisglein or @MikeHillberg do we need to take a follow up to change the documentation in the mean time?

**Comment by MikeHillberg** [comment]
I'd like to understand first what the details are.

**Comment by alexdi220** [comment]
Hello guys. Do you have plans to fix it or improve the API? I guess this should be better - there are 3 properties but only the TextAlignment works
```xaml
<TextBox HorizontalContentAlignment="Center" HorizontalTextAlignment="Center" TextAlignment="Center"/>
```

**Comment by ranjeshj** [comment]
HorizontalContentAlignment is a by product of deriving from Control, but I'm not sure if we need to have both TextAlignment and HorizontalTextAlignment. I think one of these should be removed, but this will be a breaking ABI change.
